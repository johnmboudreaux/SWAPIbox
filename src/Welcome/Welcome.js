import React from 'react';
import PropTypes from 'prop-types';

const Welcome = ({ movie }) => {

  const romanArray = ['', 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII'];

  return (
    <div className='welcome-body'>
      <div className="fade"></div>
      <section className="star-wars">

        <div className="crawl">

          <div className="title">
            <p>{ romanArray[movie.EpisodeId] }</p>
            <h1>{ movie.title }</h1>
          </div>

          <p>{ movie.openingCrawl }</p>

        </div>

      </section>
    </div>
  );
};

Welcome.propTypes = {
  compareInfo: PropTypes.objectOf(PropTypes.object),
  removeCompare: PropTypes.func,
  movie: PropTypes.object
};

export default Welcome;
