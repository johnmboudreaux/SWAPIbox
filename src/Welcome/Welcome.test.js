import React from 'react';
import { shallow } from 'enzyme';
import Welcome from './Welcome';



describe('Welcome', () => {
  it('should render correctly and match the snapshot', () => {
    const wrapper = shallow(<Welcome movie={ {
      EpisodeId: 6,
      openingCrawl:"hi",
      releaseDate: "10/12/17",
      title: "title"
    } }/>);

    expect(wrapper).toMatchSnapshot();
  });
});
