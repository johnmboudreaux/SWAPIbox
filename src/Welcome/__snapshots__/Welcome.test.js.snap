// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render correctly and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    movie={
        Object {
            "EpisodeId": 6,
            "openingCrawl": "hi",
            "releaseDate": "10/12/17",
            "title": "title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <p>
                                        VI
                              </p>
                              <h1>
                                        title
                              </h1>
                    </div>
                    <p>
                              hi
                    </p>
          </div>
</section>,
      ],
      "className": "welcome-body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <p>
                                    VI
                        </p>
                        <h1>
                                    title
                        </h1>
            </div>
            <p>
                        hi
            </p>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <p>
                                VI
                </p>
                <h1>
                                title
                </h1>
</div>,
              <p>
                hi
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    VI
</p>,
                  <h1>
                    title
</h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "VI",
                  },
                  "ref": null,
                  "rendered": "VI",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "title",
                  },
                  "ref": null,
                  "rendered": "title",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "hi",
              },
              "ref": null,
              "rendered": "hi",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <section
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <p>
                                                VI
                                    </p>
                                    <h1>
                                                title
                                    </h1>
                        </div>
                        <p>
                                    hi
                        </p>
            </div>
</section>,
        ],
        "className": "welcome-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <p>
                                          VI
                            </p>
                            <h1>
                                          title
                            </h1>
              </div>
              <p>
                            hi
              </p>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <p>
                                    VI
                  </p>
                  <h1>
                                    title
                  </h1>
</div>,
                <p>
                  hi
</p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      VI
</p>,
                    <h1>
                      title
</h1>,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "VI",
                    },
                    "ref": null,
                    "rendered": "VI",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "title",
                    },
                    "ref": null,
                    "rendered": "title",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "hi",
                },
                "ref": null,
                "rendered": "hi",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
